packages/xrpl/test/wallet/index.test.ts

```diff
@@ -297,6 +297,86 @@ describe('Wallet', function () {
     })
   })
 
+  describe('fromSecretNumbers', function () {
+    const secretNumbersString =
+      '399150 474506 009147 088773 432160 282843 253738 605430'
+    const secretNumbersArray = [
+      '399150',
+      '474506',
+      '009147',
+      '088773',
+      '432160',
+      '282843',
+      '253738',
+      '605430',
+    ]
+
+    const publicKey =
+      '03BFC2F7AE242C3493187FA0B72BE97B2DF71194FB772E507FF9DEA0AD13CA1625'
+    const privateKey =
+      '00B6FE8507D977E46E988A8A94DB3B8B35E404B60F8B11AC5213FA8B5ABC8A8D19'
+    // TODO: Uncomment when the `deriveKeypair` fix is merged so `deriveSecretNumbers` with ed25519 works.
+    // const publicKeyED25519 =
+    //   'ED8079E575450E256C496578480020A33E19B579D58A2DB8FF13FC6B05B9229DE3'
+    // const privateKeyED25519 =
+    //   'EDD2AF6288A903DED9860FC62E778600A985BDF804E40BD8266505553E3222C3DA'
+
+    it('derives a wallet using default algorithm', function () {
+      const wallet = Wallet.fromSecretNumbers(secretNumbersString)
+
+      assert.equal(wallet.publicKey, publicKey)
+      assert.equal(wallet.privateKey, privateKey)
+    })
+
+    it('derives a wallet from secret numbers as an array using default algorithm', function () {
+      const wallet = Wallet.fromSecretNumbers(secretNumbersArray)
+
+      assert.equal(wallet.publicKey, publicKey)
+      assert.equal(wallet.privateKey, privateKey)
+    })
+
+    it('derives a wallet using algorithm ecdsa-secp256k1', function () {
+      const algorithm = ECDSA.secp256k1
+      const wallet = Wallet.fromSecretNumbers(secretNumbersString, {
+        algorithm,
+      })
+
+      assert.equal(wallet.publicKey, publicKey)
+      assert.equal(wallet.privateKey, privateKey)
+    })
+
+    // TODO: Uncomment this test when the `deriveKeypair` fix is merged
+    // it('derives a wallet using algorithm ed25519', function () {
+    //   const algorithm = ECDSA.ed25519
+    //   const wallet = Wallet.fromSecretNumbers(secretNumbersString, {
+    //     algorithm,
+    //   })
+
+    //   assert.equal(wallet.publicKey, publicKeyED25519)
+    //   assert.equal(wallet.privateKey, privateKeyED25519)
+    // })
```

Comment:
This test fails because of this issue: https://github.com/XRPLF/xrpl.js/pull/2376

Commented with a TODO to re-add the test once that PR is resolved. (Open to other ideas about what's the best way to handle this :) )